gateway/test/dapi_asserts_test.go

```diff
@@ -0,0 +1,78 @@
+package test
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func requireRestSuccess(t *testing.T, resp *testHttpResponse) {
+	require.NotNil(t, resp)
+	require.Equal(t, http.StatusOK, resp.StatusCode, fmt.Sprintf("status code was not 200 body::\n%s", string(resp.Body)))
+}
+
+func requireRestSuccessNoContent(t *testing.T, resp *testHttpResponse) {
+	require.NotNil(t, resp)
+	require.Equal(t, http.StatusNoContent, resp.StatusCode, fmt.Sprintf("status code was not 202 body::\n%s", string(resp.Body)))
+}
+
+type testRestError struct {
+	Code     string
+	Resource string
+}
+
+type restErrorJson struct {
+	Code     string `json:"code,omitempty"`
+	Resource string `json:"resource,omitempty"`
+}
+
+func requireRestError(t *testing.T, resp *testHttpResponse, expectedCode int, expectedError *testRestError) {
+	require.NotNil(t, resp)
+
+	require.Equal(t, expectedCode, resp.StatusCode,
+		fmt.Sprintf("Unexpected status code %d, expected %d, body: %s",
+			resp.StatusCode, expectedCode, resp.Body))
+
+	var restErr restErrorJson
+	err := json.Unmarshal(resp.Body, &restErr)
+	require.NoError(t, err)
+
+	if expectedError != nil {
```

Comment:
Seems strange that `requireRestError` is able to be called with a nil expected error in the first place

Reply:
It's because sometimes we need to check for non-success status codes without actually validating that there is a specific response body, such as is the case with rate limiting where the rate-limiting happens ahead of any of our 'json stuff', and so the error is just text.